### PR TITLE
Properly implement status meta request

### PR DESCRIPTION
**Pull request by Others**
This is a necessary step before I can properly implement calls
into components.

Questions:
- Is all the logic in the stat tracker clear?
- How should we measure network usage?
- Should stats.rs be in the component package?

### CODE REVIEW

File: src/component/mod.rs [hunk 1]
```diff
@@ -1,22 +1,37 @@
 use std::collections::HashMap;
```
Comment (pcodes) [inline src/component/mod.rs]: Why'd you move this to a new file? What is the significance of "mod"?

File: src/stats.rs [hunk 2]
```diff
@@ -0,0 +1,121 @@
+use std::collections::VecDeque;
+use std::time::{Duration, Instant};
+
+use crate::model::ComponentStats;
+
+const DEFAULT_STAT_WINDOW: Duration = Duration::from_secs(5 * 60);
```
Comment (pcodes) [inline src/stats.rs]: Is this 5 minutes?

File: src/stats.rs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        let too_old = Instant::now() - self.stat_window;
+        while self.event_deque.front().map_or(false, |e| e.at < too_old) {
+            self.event_deque.pop_front();
+        }
+    }
+}
+
+const PERCENTILE_BUCKETS: usize = 10;
+
+fn calculate_latency_percentiles(entries: &VecDeque<StatEvent>) -> Vec<f64> {
```
Comment (pcodes) [inline src/stats.rs]: Let's chat about what this is actually getting us and why

**Comment by pcodes** [comment]
Regarding network usage, this is a tricky problem- I assume you're referring to how active the NIC is, right? If so, we'd need a way to see how much total traffic is coming in, and I don't know if there's a good way to do that in rust. I don't think we need to answer this to move forward though, we can just assume we won't be bottle-necked on the network.

**Comment by pcodes** [comment]
If stats aren't going to be used anywhere else in your project except in the component code, I'd say it makes sense to have it in the components folder.

File: src/component/mod.rs (hunk 1 above)
Comment (Others) [reply src/component/mod.rs]: It's just an organizational thing

File: src/stats.rs (hunk 2 above)
Comment (Others) [reply src/stats.rs]: Yeah unfortunately Rust doesn't have a `Duration::from_mins` method

File: src/stats.rs (hunk 3 above)
Comment (Others) [reply src/stats.rs]: Chatted in person and resolved concerns

**Pull request merged**